In [1]:
#Import Libraries
import pandas as pd
import numpy as np
import scipy.stats as stats

In [2]:
#Create DataFrame
transaction_data = [100, 150, 50, 100, 130, 120, 100, 85, 70, 150,
                   150, 120, 50, 100, 100, 140, 90, 150, 50, 90,
                   120, 100, 110, 75, 65]
df = pd.DataFrame(transaction_data, columns = ['TransactionsAmount'])

In [3]:
#Central Tendency
import statistics
print(statistics.mean(df['TransactionsAmount']))
print(statistics.median(df['TransactionsAmount']))
print(statistics.mode(df['TransactionsAmount']))

102.6
100
100


#Tulis analisismu dari Central Tendency di sini

dapat diketahui bahwa nilai mean dari data TransactionAmount bernilai 102.6, mediannya bernilai 100, dan 
nilai yang sering muncul atau modusnya bernilai 100

In [4]:
#Variability
import numpy as np
print(np.var(df['TransactionsAmount']))
print(np.std(df['TransactionsAmount']))
print(np.max(df['TransactionsAmount']),'&', np.min(df['TransactionsAmount']))
print(np.quantile(df['TransactionsAmount'], 0.25))
print(np.quantile(df['TransactionsAmount'], 0.50))
print(np.quantile(df['TransactionsAmount'], 0.75))

972.2399999999997
31.180763300471007
150 & 50
85.0
100.0
120.0


#Tulis analisismu dari Variability di sini

nilai variance dari data bernilai 972.2399999999997, dengan nilai std sebesar 31.180763300471007,
nilai tertinggi 150, sedangkan nilai terendah bernilai 50
nilai Q1 sebesar 85.0, Q2 sebesar 100.0 dan Q3 bernilai 120.0

In [5]:
#Define H0 and H1
#H0: nilai rata-rata per transaksi = $100
#H1: nilai rata-rata per transaksi > $100

In [6]:
#Define alpha = 0.05

In [7]:
#Calculate t statistics and P-value
#Perform one sample t-test
data = np.array(df['TransactionsAmount'])
pop_mean = 100
stats.ttest_1samp(data, pop_mean)

TtestResult(statistic=0.4085001556802841, pvalue=0.6865284813438117, df=24)

In [8]:
#statistic= 0.4085001556802841 , pvalue=0.6865284813438117

Conclude with alpha and p-value

Conclusion: Karena nilai p value (0.6865284813438117) lebih besar dari tingkat signifikansi (0.05), maka kita gagal menolak hipotesis nol. 
             Oleh karena itu, sales force training tidak memiliki efek yang signifikan secara statistik terhadap rata-rata per transaksi, karena nilai per transaksi masih berkisar $100

In [9]:
#Conclude with t statistics and critical region
#Find the critical region

In [10]:
#H1: lebih besar
stats.t.ppf(1-0.05, 24)

1.7108820799094275

In [11]:
#Critical region result
#karena t-statistic tidak berada pada critical regional, maka gagal menolak H0
#Conclusion: sales force training tidak efektif, rata-rata per transaksi masih berkisar $100 